src/pass/storage_rewrite.cc

```diff
@@ -491,7 +491,7 @@ class StoragePlanRewriter : public IRMutator {
   }
   // Remap the index
   Expr RemapIndex(Type dtype, Expr index, StorageEntry* e) {
-    CHECK_EQ(dtype.element_of(), e->elem_type);
+    CHECK_LE(dtype.element_of().bytes(), e->elem_type.bytes());
     if (e->elem_offset == 0) return index;
     return make_const(index.type(), e->elem_offset) + index;
```

Comment:
if bytes are different, the index calculation rule would need to change.

Reply:
say we have
a[] = 1.0 (float32)
b[] = 1  (int16)
c[] = 1  (int 16)
d[] = b[] + c[]

do you mean?
b -> a[0] elem_offset = 0
c -> a[16] elem_offset = 16